src/plugins/discourse/fetch.test.js

```diff
@@ -40,6 +40,30 @@ describe("plugins/discourse/fetch", () => {
     };
     const fetcherWithStatus = (status: number) =>
       new Fetcher(options, fakeFetch(status), 0);
+
+    it("retries on 520", async () => {
+      let nTries = 0;
+      const fetch = (url) => {
+        const status = nTries < 2 ? 520 : 200;
+        nTries++;
+        const response = new Response("", {status, url});
+        return Promise.resolve(response);
+      };
```

Comment:
This is a little more complicated than what I would call best practice for a unit test. Generally unit tests should not have any logic (Quis custodiet ipsos custodes? -- who's going to test the test?).  I'd recommend using [Jest's built in capabilities](https://jestjs.io/docs/en/mock-functions), since it's already a dependency.
```
let mockFetch = jest.fn();
// various useful functions you could use for this case:
mockFetch
  .mockReturnValueOnce(Promise.resolve(new Response("", {status: 520, "/something"})))
  .mockImplementationOnce(url => Promise.resolve(new Response("", {status: 520, url})))
  .mockReturnValue(Promise.resolve(new Response("", {status: 200, "/something"})))

const fetcher = new Fetcher(options, mockFetch, 0);
const response = await fetcher._fetch("/something");

expect(response.status).toEqual(200);
expect(mockFetch.mock.calls.length).toEqual(3);
```

Reply:
Done, thanks for the suggestion.